Could not install: `configure: error: C compiler cannot create executables`

I get this error when installing:

```
brew tap d12frosted/emacs-plus
brew install emacs-plus@29 --without-cocoa
```

```
checking whether the C compiler works... no                                          
configure: error: in `/tmp/emacs-plusA29-20230820-4164249-o3hir4/emacs-29.1':        
configure: error: C compiler cannot create executables                               
See `config.log' for more details                 
```

It seems my Ubuntu's `gcc` is too old:

```
❯ gcc --version

gcc (Ubuntu 9.3.0-17ubuntu1~20.04) 9.3.0
Copyright (C) 2019 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

❯ where gcc   
/usr/bin/gcc
```

The `gcc` installed by brew is named `gcc-13`:
```
❯ where gcc-13   
/home/linuxbrew/.linuxbrew/bin/gcc-13
```

So how do use this `gcc` for building emacs?

System Info:
```
NAME="Ubuntu"
VERSION="20.04.2 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04.2 LTS"
VERSION_ID="20.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal

5.4.0-65-generic
```

```
❯ brew config                                                                      1
HOMEBREW_VERSION: 4.1.5-55-g3cd7290
ORIGIN: https://github.com/Homebrew/brew
HEAD: 3cd72905ce3d36a84994cc6c6b9ebbdafff4e922
Last commit: 2 days ago
Core tap origin: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 7ca1c8c5509e40e60bbcb4064d935c0cd6cf1468
Core tap last commit: 3 hours ago
Core tap branch: master
Core tap JSON: 20 Aug 10:07 UTC
HOMEBREW_PREFIX: /home/linuxbrew/.linuxbrew
HOMEBREW_AUTO_UPDATE_SECS: 604800
HOMEBREW_CASK_OPTS: []
HOMEBREW_EDITOR: vim
HOMEBREW_MAKE_JOBS: 2
HOMEBREW_NO_AUTO_UPDATE: set
no_proxy: 127.0.0.1,localhost
Homebrew Ruby: 2.6.10 => /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.10_1/bin/ruby
CPU: dual-core 64-bit haswell
Clang: 16.0.6
Git: 2.39.2 => /home/linuxbrew/.linuxbrew/bin/git
Curl: 7.68.0 => /usr/bin/curl
Kernel: Linux 5.4.0-65-generic x86_64 GNU/Linux
OS: Ubuntu 20.04.2 LTS (focal)
Host glibc: 2.31
/usr/bin/gcc: 9.3.0
/usr/bin/ruby: 2.7.0
glibc: 2.35_1
gcc@11: 11.4.0
gcc: 13.1.0
xorg: N/A
```

--- a/Formula/emacs-plus@28.rb
+++ b/Formula/emacs-plus@28.rb
@@ -5,7 +5,10 @@ class EmacsPlusAT28 < EmacsBase
   url "https://ftp.gnu.org/gnu/emacs/emacs-28.2.tar.xz"
   mirror "https://ftpmirror.gnu.org/emacs/emacs-28.2.tar.xz"
   sha256 "ee21182233ef3232dc97b486af2d86e14042dbb65bbc535df562c3a858232488"
-  env :std
+
+  on_macos do
+    env :std
+  end
 
   desc "GNU Emacs text editor"
   homepage "https://www.gnu.org/software/emacs/"

--- a/Formula/emacs-plus@29.rb
+++ b/Formula/emacs-plus@29.rb
@@ -5,7 +5,10 @@ class EmacsPlusAT29 < EmacsBase
   url "https://ftp.gnu.org/gnu/emacs/emacs-29.3.tar.xz"
   mirror "https://ftpmirror.gnu.org/emacs/emacs-29.3.tar.xz"
   sha256 "c34c05d3ace666ed9c7f7a0faf070fea3217ff1910d004499bd5453233d742a0"
-  env :std
+
+  on_macos do
+    env :std
+  end
 
   desc "GNU Emacs text editor"
   homepage "https://www.gnu.org/software/emacs/"

--- a/Formula/emacs-plus@30.rb
+++ b/Formula/emacs-plus@30.rb
@@ -3,7 +3,10 @@ require_relative "../Library/EmacsBase"
 class EmacsPlusAT30 < EmacsBase
   init 30
   version "30.0.50"
-  env :std
+
+  on_macos do
+    env :std
+  end
 
   desc "GNU Emacs text editor"
   homepage "https://www.gnu.org/software/emacs/"

